rust/crd/src/lib.rs

```diff
@@ -122,10 +127,51 @@ pub struct NifiStatus {}
 #[serde(rename_all = "camelCase")]
 pub struct NifiConfig {
     pub log: Option<NifiLogConfig>,
+    pub resources: Option<Resources<NifiStorageConfig, NoRuntimeLimits>>,
 }
 
 impl NifiConfig {
     pub const NIFI_SENSITIVE_PROPS_KEY: &'static str = "NIFI_SENSITIVE_PROPS_KEY";
+
+    pub fn default_resources() -> Resources<NifiStorageConfig, NoRuntimeLimits> {
+        Resources {
+            memory: MemoryLimits {
+                limit: Some(Quantity("6Gi".to_string())),
+                runtime_limits: NoRuntimeLimits {},
+            },
+            cpu: CpuLimits {
+                min: Some(Quantity("1".to_string())),
```

Comment:
This is a pretty huge reservation that's probably going to be problematic for laptop-scale clusters.

Reply:
I agree. I would have preferred for the default resources to be `None`.

Reply:
I agree. I would have preferred for these to be `None`.

Reply:
Might want to coordinate this a bit, looks like Malte's kafka-op PR does indeed ship without default limits: https://github.com/stackabletech/kafka-operator/pull/405/files#diff-b10253ea28ef51076309f375e97cd5e2795500b0e07c5326c413055bffcb4a85R145-R163

Reply:
Might want to coordinate this a bit, looks like @maltesander's kafka-op PR does indeed ship without default limits: https://github.com/stackabletech/kafka-operator/pull/405/files#diff-b10253ea28ef51076309f375e97cd5e2795500b0e07c5326c413055bffcb4a85R145-R163